protocols/kad/src/behaviour.rs

```diff
@@ -912,9 +912,17 @@ where
     /// The result of this operation is delivered in a
     /// reported via [`KademliaEvent::OutboundQueryCompleted{QueryResult::GetProviders}`].
     pub fn get_providers(&mut self, key: record::Key) -> QueryId {
+        let providers = self
+            .store
+            .providers(&key)
+            .into_iter()
+            .filter(|p| !p.is_expired(Instant::now()))
+            .map(|p| p.provider)
+            .collect();
+        // TODO(stn): call remove_provider on expired providers?
```

Comment:
```suggestion
```

I just double checked, expired provider records are periodically garbage collected (see below). Thus calling `remove_provider` here is not needed.

https://github.com/libp2p/rust-libp2p/blob/e83e1b3d0be822f58d705bc724c87f267c806150/protocols/kad/src/jobs.rs#L308-L314